src/mlpack/methods/matrix_completion/mc_fw_update_matrix.hpp

```diff
@@ -0,0 +1,238 @@
+/**
+ * @file mc_fw_update_matrix.hpp
+ * @author Chenzhe Diao
+ *
+ * In FrankWolfe algorithm, used as UpdateRuleType.
+ * Designed for matrix completion problems.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_MATRIX_COMPLETION_MC_FW_UPDATE_MATRIX_HPP
+#define MLPACK_METHODS_MATRIX_COMPLETION_MC_FW_UPDATE_MATRIX_HPP
+
+#include "mc_fw_function.hpp"
+#include <mlpack/core/optimizers/proximal/proximal.hpp>
```

Comment:
Do you mind to move the ``Proximal`` class into the ``fw`` folder? I think it would improve the overall code structure since the ``Proximal`` class isn't directly an optimizer. Let me know what you think.

Reply:
Thanks for the review. Sorry for the late reply since I was busy last weekend. This is actually the idea of @stephentu  to make a `Proximal` class folder like this. Since Proximal Algorithms are a large group of optimization algorithms based on proximal operators, and the projections I implemented here are 2 cases of them, I don't know if any one would add more algorithms of this type in the future. Since it is not directly an optimizer here, I think it is ok for me to move them into `fw` folder for now. I will leave a message to Stephen to see what he thinks about it. :D

Reply:
Hi @zoq. The idea was, if anybody ever comes along and need prox operators or projection operators (ie if they implement projected gradient descent), then they can reuse these operators. Perhaps the naming is not the best, but I do think there is value in keeping it in a separate location to be shared.

Reply:
Thanks for the clarification, I guess if someone is going to implement e.g. projected gradient descent as you said, it would make sense to put the proximal operations inside the projected gradient descent optimizer folder. My point is, right now, each folder is basically an optimizer that follows some kind of unified interface; we can directly use each optimizer in combination with some existing method. The proximal folder is an exception. Not sure that's something we should worry about.